VisualPinball.Unity/VisualPinball.Unity.Editor/Utils/TreeViewWithTreeModel/TreeElement.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+namespace VisualPinball.Unity.Editor.Utils.TreeViewWithTreeModel
+{
+
+	/// <summary>
+	/// TreeElement is the base class for creating TreeModel which will be used by TreeViewWithTreeModel, you'll have to inherit from it.
+	/// It provides base properties to be inserted & identified in a TreeView (Id, Name, Depth)
+	/// It hosts also its hierarchical information.
+	/// Some properties are serializable to save tree structure as Assets.
+	/// </summary>
+	[Serializable]
+	public class TreeElement
+	{
+		[SerializeField]
+		public int Id { get; set; }
+		[SerializeField]
+		public virtual string Name { get; }
+
+		[SerializeField]
+		public int Depth { get; set; }
+
+		[NonSerialized]
+		private TreeElement _parent;
+		public TreeElement Parent { get { return _parent; } set { _parent = value; } }

```

Comment:
Any particular reason to make this a property and not just a public (non-serialized) field? Same question for `Children` below

Reply:
nope except it came from the Unity TreeViewWithModel sample :)

Reply:
Yeah some people's idea of OOP is that all fields should be private. However I fail to see the advantage (in most cases, anyway) but only the boiler plate :)